src/core/src/service_interfaces/LifecycleManagerAzure.py

```diff
@@ -41,52 +41,42 @@ def execution_start_check(self):
-                            current_time = self.env_layer.datetime.datetime_utcnow()
-                            elapsed_time_in_minutes = self.env_layer.datetime.total_minutes_from_time_delta(current_time - timer_start_time)
-                            if elapsed_time_in_minutes > Constants.REBOOT_BUFFER_IN_MINUTES:
-                                self.composite_logger.log_debug("Auto-assessment is now considered safe to start since Core did not start after a reboot buffer wait period. [LastHeartbeat={0}][Operation={1}]".format(str(core_sequence['lastHeartbeat']), str(core_sequence['action'])))
-                                break
-
-                        self.update_core_sequence(completed=False)  # signalling core restart with auto-assessment as its safe to do so
+                    self.composite_logger.log_debug("Auto-assessment is SAFE to start. Existing sequence number marked as completed.")
+                    break
+
+                # Check for active running processes if not completed
+                if len(self.identify_running_processes(core_sequence['processIds'])) != 0:
+                    # NOT SAFE TO START. Possible reasons: full core operation is in progress (okay), some previous auto-assessment is still running (bad scheduling, adhoc run, or process stalled)
+                    self.composite_logger.log_warning("Auto-assessment is NOT safe to start yet. Existing core process(es) running. Exiting. [LastHeartbeat={0}][Operation={1}]".format(str(core_sequence['lastHeartbeat']), str(core_sequence['action'])))
+                    self.env_layer.exit(0)
+
+                # Safely timeout if wait for any core restart events (from a potential reboot) has exceeded the maximum reboot buffer
+                current_time = self.env_layer.datetime.datetime_utcnow()
+                elapsed_time_in_minutes = self.env_layer.datetime.total_minutes_from_time_delta(current_time - timer_start_time)
+                if elapsed_time_in_minutes > Constants.REBOOT_BUFFER_IN_MINUTES:
+                    self.composite_logger.log_debug("Auto-assessment is now considered SAFE to start since Core did not start after a reboot buffer wait period. [LastHeartbeat={0}][Operation={1}]".format(str(core_sequence['lastHeartbeat']), str(core_sequence['action'])))
+                    break
+
+                # Briefly pause execution to re-check all states (including reboot buffer) again
+                self.composite_logger.log_debug("Auto-assessment is waiting for up to the Core reboot-buffer period. [LastHeartbeat={0}][Operation={1}][ElapsedTimeInMinutes={2}][TotalWaitTimeRequired={3}]".format(str(core_sequence['lastHeartbeat']), str(core_sequence['action']), str(elapsed_time_in_minutes), str(Constants.REBOOT_BUFFER_IN_MINUTES)))
+                time.sleep(15)
+
+            # Signalling take-over of core state by auto-assessment after safety checks for any competing process
+            self.update_core_sequence(completed=False)
```

Comment:
Need to make these changes in LifeCycleManagerArc.py too

Reply:
Leaving that to IDC since it requires validation with Arc